blint/db.py

```diff
@@ -0,0 +1,232 @@
+
+    # Errors not being caught here
+    output_list = get_bid_using_ename_batch(symbols_list)
+
+    eid_list = [it[0] for it in output_list]
+    bid_2d_list = [it[1] for it in output_list]
+    for it in range(len(eid_list)):
+        bid_list = bid_2d_list[it].split(",")
+        score = 1 / len(bid_list)
+        for bid in bid_list:
+            bname = get_bname(bid)
+            if bname in binaries_detected_dict:
+                binaries_detected_dict[bname] += score
+            else:
+                binaries_detected_dict[bname] = score
+    return binaries_detected_dict
+
+
+def get_bid_using_ename_batch(batch_export_name):
+    """Retrieves binary IDs associated with a batch of export names from a SQLite database.
+
+    This function queries the database to find binary IDs linked to specific export names in a batch.
+    It uses the BLINTDB_LOC environment variable to connect to the SQLite database and performs a grouped query.
+
+    Args:
+        batch_export_name (list): A list of export names to search for in the database.
+
+    Returns:
+        list: A list of tuples containing export IDs and their corresponding concatenated binary IDs.
+
+    Raises:
+        sqlite3.Error: If there are any database connection or query execution issues.
+    """
+    BLINTDB_LOC = os.getenv("BLINTDB_LOC")
+
+    
+
+    with closing(sqlite3.connect(BLINTDB_LOC)) as connection:
+        with closing(connection.cursor()) as c:
+            place_holders = "?, " * (len(batch_export_name) - 1) + "?"
+            if len(batch_export_name) > 0:
+                output_string = f"SELECT eid, group_concat(bid) from BinariesExports where eid IN (SELECT rowid from Exports where infunc IN ({place_holders})) group by eid"
+                # print(output_string)
+                c.execute(output_string, batch_export_name)
+            res = c.fetchall()
+        connection.commit()
+    return res
+
+
+def get_bname(bid):
+    """Retrieves the binary name for a given binary ID from a SQLite database. The function performs a lookup to fetch the corresponding binary name.
+
+    This function connects to a SQLite database using the BLINTDB_LOC environment variable and executes a query to retrieve the binary name based on the provided binary ID. If no matching binary is found, it returns None.
+
+    Args:
+        bid (int): The binary identifier to search for in the database.
+
+    Returns:
+        str or None: The name of the binary if found, otherwise None.
+
+    Raises:
+        sqlite3.Error: If there are any database connection or query execution issues.
+    """
+    BLINTDB_LOC = os.getenv("BLINTDB_LOC")
+    with closing(sqlite3.connect(BLINTDB_LOC)) as connection:
+        with closing(connection.cursor()) as c:
+            c.execute("SELECT bname from Binaries where bid=?", (bid,))
+            res = c.fetchall()
+        connection.commit()
```

Comment:
Can you remove all commit statements? Also, can you change the logic to reuse connections instead of closing them after every invocation?